### DISCUSSION TITLE
Github Action Not Showing

### DISCUSSION THREAD

**Discussion by rogebrd** [community/community#26479] [Actions]
<p>I am trying to create an action that publishes to pypi when a release is created.  I am not seeing the action show up on the action tab though.  Does that mean there is a bug in the action code or am I missing somehting else?</p>
<p>Repo: <a href="https://github.com/dropbox/dropbox-sdk-python" rel="nofollow noopener">https://github.com/dropbox/dropbox-sdk-python</a></p>
<p>Workflow: <a href="https://github.com/dropbox/dropbox-sdk-python/blob/master/.github/workflows/pypiupload.yml" rel="nofollow noopener">https://github.com/dropbox/dropbox-sdk-python/blob/master/.github/workflows/pypiupload.yml</a></p>

**Comment by mscoutermarsh** [comment]
<p>Hi <a class="mention" href="/rogebrd">@rogebrd</a>,</p>
<p>This all looks correct to me. It will show up in the Actions tab after the first run.</p>
<p>It should have started when your last release was created. We did have an incident yesterday &amp; it’s possible the event was dropped (unlikely, but we should rule that possibility out). Could you try creating another release and see if it runs?</p>
<p>Thanks. </p>

**Comment by rogebrd** [comment]
<p>Appears to be working now.  I am noticing that this section of code appears to not be working:</p>
<pre><code>- name: Build Sources
      run: |
        python setup.py sdist
      if: ${{ matrix.python-version }} == '3.x'
</code></pre>
<p>Specifically the if section of this.  It appears to have ran for both python 2.7 and 3.x.  Do you know if there is any issue with my syntax?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/rogebrd">@rogebrd</a> ,</p>
<p>Yeah, you use an incorrect syntax for the <strong>if conditional</strong> , reference the description from the docs about <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#about-contexts-and-expressions" rel="nofollow noopener"><strong>Context and expression syntax for GitHub Actions</strong></a>:</p>
<blockquote>
<p>When you use expressions in an if conditional, you do not need to use the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression.</p>
</blockquote>
<p>So the correct syntax is:</p>
<pre><code>- name: Build Sources
      if: matrix.python-version == '3.x'
      run: |
        python setup.py sdist
</code></pre>
<p>In addition, a summary for your first question:</p>
<p>After you have created a new workflow, this workflow will not be listed on " <strong>Actions</strong>" tab until it is triggered for the first run.</p>

**Comment by rogebrd** [comment]
<p>Great, I must have missed something when I was reading through the documentation.  Thanks for your help</p>

**Comment by lc0** [comment]
<p>Seems like I am having the same issue.</p>
<p>I created a workflow from template - pushed two tag releases and still nothing on actions page - <a href="https://github.com/lc0/altair-images" rel="nofollow noopener">https://github.com/lc0/altair-images</a></p>
<p>Am I missing something obvious? </p>
